scripts/ynh_composer__2

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Execute a command with Composer
+#
+# usage: ynh_composer_exec --phpversion=phpversion [--workdir=$final_path] --commands="commands"
+# | arg: -w, --workdir - The directory from where the command will be executed. Default $final_path.
+# | arg: -c, --commands - Commands to execute.
+ynh_composer_exec () {
+	# Declare an array to define the options of this helper.
+	local legacy_args=vwc
+	declare -Ar args_array=( [v]=phpversion= [w]=workdir= [c]=commands= )
+	local phpversion
+	local workdir
+	local commands
+	# Manage arguments with getopts
+	ynh_handle_getopts_args "$@"
+	workdir="${workdir:-$final_path}"
+	phpversion="${phpversion:-7.0}"
+
+	COMPOSER_HOME="$workdir/.composer" \
+		php${phpversion} "$workdir/composer.phar" $commands \
+		-d "$workdir" --quiet --no-interaction
+}
+
+# Install and initialize Composer in the given directory
+#
+# usage: ynh_install_composer --phpversion=phpversion [--workdir=$final_path]
+# | arg: -w, --workdir - The directory from where the command will be executed. Default $final_path.
+ynh_install_composer () {
+	# Declare an array to define the options of this helper.
+	local legacy_args=vw
+	declare -Ar args_array=( [v]=phpversion= [w]=workdir= )
+	local phpversion
+	local workdir
+	# Manage arguments with getopts
+	ynh_handle_getopts_args "$@"
+	workdir="${workdir:-$final_path}"
+	phpversion="${phpversion:-7.0}"
+
+	curl -sS https://getcomposer.org/installer \
+		| COMPOSER_HOME="$workdir/.composer" \
+		php${phpversion} -- --quiet --install-dir="$workdir" \
+		|| ynh_die "Unable to install Composer."
+
+	# update dependencies to create composer.lock
```

Comment:
Why do you run `composer install --no-dev` here?
Shouldn't that command not only install composer and nothing else?

Reply:
It's an helper, it's made like that, just to install composer.
the helper is used at:
https://github.com/YunoHost-Apps/kimai2_ynh/blob/b8f47351c6d8527b180f44df738ff297278fc213/scripts/install#L156-L158

Reply:
Yes, I have seen that. I just wonder why you execute "composer install" after installation, because it is executed later on. Isn't that an unnecessary call at this place?

Reply:
I wasn't sure if it was necessary or not. I added it to have the `--optimize-autoloader` that is requested in the documentation.

Should I remove ? https://github.com/YunoHost-Apps/kimai2_ynh/blob/b8f47351c6d8527b180f44df738ff297278fc213/scripts/install#L157

Reply:
I would remove it here, as the single responsibility of this helper function is to install composer, right?
And then keep it in the install and upgrade scripts as they are.

Reply:
as I prefer to keep the helper as it is... or maybe doing a PR to the helper ....

Reply:
helper modified to integrate additional arguments